test/mason/run/mason-run.chpl

```diff
@@ -0,0 +1,6 @@
+use MasonRun;
+
+proc main() {
+  const args = ["mason", "run"];
```

Comment:
It looks like the new test depends on the `FizzBuzz` target being built. I'd suggest adding the `--build` flag to accomplish that.

Reply:
The usage for `mason run` says it will build the project if needed. Also, the masonRun() code does call masonBuild() when lockfile is not present.

* https://github.com/thomasvandoren/chapel/blob/8817-mason-run/tools/mason/MasonHelp.chpl#L77-L79
* https://github.com/thomasvandoren/chapel/blob/a3ac4f295505d5ef5b3ad682ba82c8c1819ac11f/tools/mason/MasonRun.chpl#L59-L64

Did you see a test failure for this?

Reply:
Ah, we're definitely exposing some issues here.

> Did you see a test failure for this?

I did. Maybe you are not seeing it because `target/` exists locally?

> `writeln('   - If no target directory, build and run is Mason.toml is present');`

This documentation has a typo and is not the current behavior. The conditional block you linked depends on `target/` existing in the first place:

-
 https://github.com/thomasvandoren/chapel/blob/a3ac4f295505d5ef5b3ad682ba82c8c1819ac11f/tools/mason/MasonRun.chpl#L50

A simple fix would be to update that line of the help usage, but I suspect that is not the behavior we want...

Reply:
Ok, got it. I'll update test to include `--build` flag, and file issue for usage/mason run behavior when target dir is not present. Thanks!

Reply:
That would be great, thanks! I think it will be ready to merge after that commit.

Reply:
I updated the code and filed this issue: https://github.com/chapel-lang/chapel/issues/8852